support-frontend/assets/components/subscriptionCheckouts/address/addressFields.tsx

```diff
@@ -88,27 +91,24 @@ export function AddressFields({ scope, ...props }: PropTypes): JSX.Element {
 			</Select>
 			{props.country === 'GB' ? (
 				<PostcodeFinder
-					{...props.postcodeState}
-					setPostcode={props.setPostcodeForFinder}
-					fetchResults={props.fetchResults}
-					id={`${scope}-postcode-lookup`}
-					onPostcodeUpdate={props.setPostcode}
-					onAddressUpdate={({
+					postcode={props.postcodeState.postcode}
+					isLoading={props.postcodeState.isLoading}
+					error={props.postcodeState.error}
+					results={props.postcodeState.results}
+					onPostcodeUpdate={(postcode) => {
+						props.setPostcode(postcode);
+						props.setPostcodeForFinder(postcode);
+					}}
+					onPostcodeError={props.setPostcodeErrorForFinder}
+					onFindAddress={props.onFindAddress}
+					onAddressSelected={({
 						lineOne,
 						lineTwo,
 						city,
 					}: PostcodeFinderResult) => {
-						if (lineOne) {
-							props.setLineOne(lineOne);
-						}
-
-						if (lineTwo) {
-							props.setLineTwo(lineTwo);
-						}
-
-						if (city) {
-							props.setTownCity(city);
-						}
+						props.setLineOne(lineOne ?? '');
+						props.setLineTwo(lineTwo ?? '');
+						props.setTownCity(city ?? '');
```

Comment:
I removed the conditional checking because of an edge case I encountered. If you first select an address that has a `lineTwo` and then select another address that doesn't, the old `lineTwo` doesn't get overwritten and is left in the form